x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/HashAggregationOperator.java

```diff
@@ -115,39 +134,61 @@ public void add(int positionOffset, IntVector groupIds) {
                 }
             }
         });
+        if (chunkOutput && blockHash.size() >= maxPageSize) {
+            output = emitPage();
```

Comment:
OH! This isn't what I expected.

I had thought we'd buffer up all the data at once and then cut on the way out. And at some point we'd spill the state to disk. So we'd only ever build a single block hash still. Is that still a good idea?

Reply:
Thanks, Nik! Your proposal is safer than mine in this PR because we will avoid hashing keys multiple times. The downside is that we will need more memory for the BlockHash. I will update this PR according to your suggestion, and we will reconsider this (or any other option) when we can spill to disk.